index.ts

```diff
@@ -0,0 +1,74 @@
+import { EventEmitter } from 'events';
+
+type Handler = (...args: any[]) => void;
+type EventMap = {
+  [k: string]: Handler | Handler[] | undefined;
+};
+
+function safeApply<T, A extends any[]> (handler: (this: T, ...args: A) => void, context: T, args: A): void {
+  try {
+    Reflect.apply(handler, context, args);
+  } catch (err) {
+    // Throw error after timeout so as not to interrupt the stack
+    setTimeout(() => {
+      throw err;
+    });
+  }
+}
+
+function arrayClone<T> (arr: T[]): T[] {
```

Comment:
Isn't this function the same as `arr.slice()` or `[...arr]`? Doesn't seem necessary to include.